src/Core/GetCandy.php

```diff
@@ -63,6 +65,18 @@ public function setIsHubRequest($bool)
         return $this;
     }
 
+    public static function onChannel($channel, \Closure $closure)
+    {
+        $current = FetchCurrentChannel::run();
+        SetCurrentChannel::run([
```

Comment:
Why do you need to set it here?

Reply:
We're setting it before the closure runs and then resetting it afterwards.

Reply:
Ah I'm reading it wrong! Carry on! :D